app/models/vpago/payment_decorator.rb

```diff
@@ -3,6 +3,7 @@ module PaymentDecorator
     def self.prepended(base)
       base.has_many :payouts, class_name: 'Spree::Payout', inverse_of: :payment
       base.after_create -> { Vpago::PayoutsGenerator.new(self).call }, if: :should_generate_payouts?
+      base.after_create :mark_as_completed, if: -> { auto_capture? && ticket_seller_user? }
```

Comment:
Avoid adding the `ticket_seller_user?` logic directly here. This is an abstract payment method and should remain generic, not specific to `ticket_seller`. Instead, let's define a `payment_received?` method as shown below:

```ruby
base.after_create :mark_as_completed, if: -> { auto_capture? && payment_received? }
```

By default, this method should be:

```ruby
def payment_received?
  false
end
```

Then, in the `Commissioner`, override the `payment_received?` method with the specific logic you've defined for `ticket_seller`. This keeps the abstraction clean and ensures flexibility for other use cases.

Add in a README how the received payment work ( direct sale with case )